pkg/kubelet/status/manager.go

```diff
@@ -182,16 +195,42 @@ func (m *manager) TerminatePods(pods []*api.Pod) bool {
 				Terminated: &api.ContainerStateTerminated{},
 			}
 		}
-		select {
-		case m.podStatusChannel <- podStatusSyncRequest{pod, pod.Status}:
-		default:
+		newStatus := m.updateStatusInternal(pod, pod.Status)
+		if newStatus != nil {
+			select {
+			case m.podStatusChannel <- podStatusSyncRequest{pod.UID, *newStatus}:
```

Comment:
Do we still need to hold the lock while sending the request to the channel? I think we no longer have to do this, similar to what SetPodStatus does now.